File: python/leetcode/LEETCODE_141.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        p: Optional[ListNode] = head
        visited: List[int] = []
            
        while p:
            p = p.next

            if id(p) in visited:
                return True
            else:
                visited.append(id(p))
                
        return False

File: java/src/leetcode/LEETCODE_141.java
package leetcode;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class LEETCODE_141 {
    public boolean hasCycle(ListNode head) {
        ListNode p = head;
        List<ListNode> visited = new ArrayList<>();
        
        while (!Objects.isNull(p)) {
            p = p.next;
            
            if (visited.contains(p)) {
                return true;
            } else {
                visited.add(p);
            }
        }
        
        return false;
        
    }
}
